RecursiveExtractor/Extractors/WimExtractor.cs

```diff
@@ -107,8 +104,12 @@ public IEnumerable<FileEntry> Extract(FileEntry fileEntry, ExtractorOptions opti
             {
                 for (var i = 0; i < baseFile.ImageCount; i++)
                 {
-                    var image = baseFile.GetImage(i);
-                    foreach (var file in image.GetFiles(image.Root.FullName, "*.*", SearchOption.AllDirectories))
+                    if (!TryGetImage(baseFile, i, out var image))
```

Comment:
DiscUtils returns a `WimFile` with `ImageCount` == 1, but there doesn't appear to be any valid `WimFileSystem` that can be retrieved. As far as I can see that's not due to a bug in DiscUtils, it's just doing the best it can with invalid data.

I'm totally sure how you would want to handle an image that can't be retrieved, but this is my proposal.

Reply:
I think this should respect the ExtractSelfOnFail argument, see here for an example of a similar handler for zips. https://github.com/microsoft/RecursiveExtractor/blob/ac94200eeb62942213bb9ac0e88624c934e4e496/RecursiveExtractor/Extractors/ZipExtractor.cs#L66

I can add that to the PR.

Reply:
I think this approach is fine. I have added a debug message for when an image fails to extract. The other approach I can think of is to set failed archive and extract self on fail if all images fail to extract from a WIM but in the cases that a  partial extraction succeeds I think we generally err on the side of best effort extraction of as much as possible.

Reply:
ok sounds good!